generator.ts

```diff
@@ -0,0 +1,217 @@
+import { Confirm, Input } from "@webpack-cli/webpack-scaffold";
+import Generator from "yeoman-generator";
+import createDevConfig from "./dev-config";
+import questions from "./questions";
+import getPackageManager from "./utils/package-manager";
+import welcomeMessage from "./utils/welcome";
+
+module.exports = class WebpackGenerator extends Generator {
+ [x: string]: any;
+ public dependencies: string[];
+ public devConfig: any;
+ public webpackOptions: any;
+ public manifestDetails: any;
+ constructor(args, opts) {
+  super(args, opts);
+  this.dependencies = ["webpack"];
+  opts.env.configuration = {
+   dev: {
+    manifestDetails: {},
+    webpackOptions: {},
+   },
+  };
+  this.devConfig = {
+            manifestDetails: {},
+            webpackOptions: {},
+        };
+  this.webpackOptions = this.devConfig.webpackOptions;
+  this.manifestDetails = this.devConfig.manifestDetails;
+ }
+
+ /**
+  * Prompts user for actions on CLI
+  * It uses prompt() method from Inquirer.js
+  * @returns {void}
+  */
+ public prompting(): Promise<void> {
+
+  const done = this.async();
+  let serviceWorker = false;
+  let manifestDetails = {};
+  let favPath;
+  let outputDir;
+
```

Comment:
I am not sure, but I think that these variable declarations should also have types.

The rationale behind it is that suppose we expect `favPath` to be a string, but it gets reassigned as `boolean` mistakenly.  So this `TS` system should be able to flag that as an error.

Reply:
Here we should 😁
Thanks @ajayposhak 👍